.travis.yml

```diff
@@ -37,17 +40,54 @@ env:
 matrix:
   fast_finish: true
   include:
+# PHP versions compatibility (latest WordPress and most used WooCommerce versions)
   - name: "Coding standard check"
-    php: 7.2
-    env: WP_VERSION=latest WC_VERSION=4.3.0 RUN_PHPCS=1
+    php: 7.0
+    env: WP_VERSION=latest WC_VERSION=4.5.0 RUN_PHPCS=1
+    stage: test
+  - php: 7.1
+    env: WP_VERSION=latest WC_VERSION=4.5.0
+    stage: test
+  - php: 7.2
+    env: WP_VERSION=latest WC_VERSION=4.5.0
     stage: test
   - php: 7.3
-    env: WP_VERSION=latest WC_VERSION=4.3.0
+    env: WP_VERSION=latest WC_VERSION=4.5.0
     stage: test
-  - name: "PHP 7.3 unit tests using WordPress nightly"
-    php: 7.3
-    env: WP_VERSION=nightly WC_VERSION=4.3.0
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=4.5.0
     stage: test
+# bottom compatibility edge (oldest supported PHP/WordPress/WooCommerce versions)
+  - php: 7.0
+    env: WP_VERSION=5.3 WC_VERSION=4.0.0
+    stage: compatibility
+# WooCommerce versions compatibility (latest supported PHP/WordPress versions on latest supported PHP version)
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=4.0.0
+    stage: compatibility
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=4.1.0
+    stage: compatibility
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=4.2.0
+    stage: compatibility
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=4.3.0
+    stage: compatibility
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=4.4.0
+    stage: compatibility
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=4.5.0
+    stage: compatibility
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=4.6.0
+    stage: compatibility
+# top compatibility edge (nightly versions of WordPress on latest supported WooCommerce/PHP version)
+  - php: 7.4
+    env: WP_VERSION=nightly WC_VERSION=latest
+    stage: compatibility
```

Comment:
How about moving this one into own stage? I expect this will fail from time to time which is acceptable but would be nice to see it explicitly.

Reply:
Ok, will move to a separate stage. When it comes to failures, do you mean that we should allow it to fail (I'd say it makes sense)?

Reply:
I'd say yes, because with nightly builds there might be failures not related to PR. They should be checked and investigated but might be fixed as part of different PR/issue.